packages/editor/src/lib/editor/tools/StateNode.ts

```diff
@@ -73,6 +74,8 @@ export abstract class StateNode implements Partial<TLEventHandlers> {
 	initial?: string
 	children?: Record<string, StateNode>
 	parent: StateNode
+	// Flag that indicates whether the `StateNode` contains some changes that have not been flushed to the Editor.
+	isDirty: boolean
```

Comment:
I'm a bit wary of making this public. Seems like an implementation detail and I think I'd rather keep it private or internal and related to pointer movement specifically for now until we see a wider pattern develop.